### ISSUE TITLE
Fix config files

### ISSUE THREAD

**Issue by znicholls** [MESMER-group/mesmer#34]
The config files are currently raw Python files. We should find a better way to do this.

**Comment by mathause** [comment]
Currently a config state is passed as a param to each function. Also there is no default configuration (or is there?). Can we avoid `cfg` entirely? If not here are some projects where we could get inspiration from:

- https://docs.oggm.org/en/stable/getting-started.html
- https://docs.dask.org/en/stable/configuration.html

Of course the difficult part is to decide which info should be part of the config and what is passed as arguments.

---

copied this comment from #112 to avoid having a duplicate issue

**Comment by mathause** [comment]
Some 'config values' I started using. 


```
x_dim="lon"
y_dim="lat"
x_coords=None # -> x_dim (important for RCMs)
y_coords=None # -> y_dim (important for RCMs)
cell_dim="cell" # not a good name
time_dim="time"
```

**Comment by yquilcaille** [comment]
Why having "x_coords" and "x_dim"? Is it called differently in RCMs?

And instead of "cell", why not gridcell? A little longer, but more explicit?

**Comment by mathause** [comment]
> Why having "x_coords" and "x_dim"? Is it called differently in RCMs?

Yes, the coords can be different from the dimensions. 

```python
rasm = xr.tutorial.open_dataset("rasm")

```

`x` and `y` are the dims and `xc` and `yc` are the coords


> And instead of "cell", why not gridcell? A little longer, but more explicit?

No super strong opinion - we leaned towards `cell` in  #105 but I am happy to reconsider.

**Comment by yquilcaille** [comment]
Indeed, I remember about the differences between coordinates and dimensions :) But I was rather wondering why this mention to RCMs here? Is it different for them? I would have expected that it would have been the same for RCMs and ESMs?

Yes, `gp` and `cell` were mentionned, but you were saying it is not a good name. Hence `gridcell`, but i suppose others may have new ideas?

**Comment by mathause** [comment]
Yes RCMs are usually on a rotated grid.

**Comment by mathause** [comment]
We moved away from the global config. Not sure we replaced it with something better but here we are...

**Comment by veni-vidi-vici-dormivi** [comment]
Hey I think we did...

**Comment by mathause** [comment]
(Yes - I was thinking about ways to calibrate the whole ensemble which we just do in code now. Instead of setting the models in a separate file...)
